code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,909 @@
+	playsound(src, 'sound/items/welder.ogg', 150, TRUE)
+	qdel(src)
+
+
+///nearly all runes use their turf in some way so we set rune_turf to their turf automatically, the rune also start on cooldown if it uses one.
+/obj/effect/warped_rune/Initialize()
+	. = ..()
+	rune_turf = get_turf(src)
+	RegisterSignal(rune_turf, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+	cooldown = world.time + max_cooldown
+
+
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+
+///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(isturf(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
+		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
+		return
+
+	if(istype(target, runepath))       //checks if the target is a rune and then if you can store it
+		if(warp_charge >= max_charge)
+			to_chat(user, "<span class='warning'>Your [src] is already full!</span>")
+			return
+
+		if(do_after(user, storing_time,target = target) && warp_charge < max_charge)
+			to_chat(user, "<span class='notice'>You store the rune in [src].</span>")
+			qdel(target)
+			warp_charge++
+			desc = "It just won't stay in place. it has [src.warp_charge] charge left"
+			return
+
+	if(!istype(target,/turf/open))
+		return
+
+	if(isspaceturf(target))
+		to_chat(user, "<span class='warning'>you cannot draw a rune in space!</span>")
+		return
+
+	if((istype(target,/turf/open/lava)) || (locate(/obj/structure) in target) || (istype(target,/turf/open/chasm))) // check if there's a wall or a structure in the way
+		to_chat(user, "<span class='warning'>You can't draw the rune here!</span>")
+		return
+
+	if(warp_charge > 0 && do_after(user, drawing_time,target = target)) //spawns the right rune if you have charge(s) left
+		if(warp_charge <= 0) //this is only here to fix a bug where the user can draw multiple runes at once if they spam click
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+			return
+
+		playsound(target, 'sound/effects/slosh.ogg', 20, TRUE)
+		warp_charge--
+		desc = "It just won't stay in place. it has [src.warp_charge] charge left"
+		new runepath(target)
+		to_chat(user, "<span class='notice'>You carefully draw the rune with [src].</span>")
+
+
+/obj/item/slimecross/warping/grey
+	name = "greyspace crossbreed"
+	colour = "grey"
+	effect_desc = "Creates a rune. Extracts that are on the rune are absorbed, 8 extracts produces an adult slime of that color."
+	runepath = /obj/effect/warped_rune/greyspace
+
+
+/obj/effect/warped_rune/greyspace
+	name = "greyspace rune"
+	desc = "Death is merely a setback, anything can be rebuilt given the right components"
+	icon_state = "greyspace_rune"
+	max_cooldown = 50
+	///number of slime extract currently absorbed by the rune
+	var/absorbed_extract = 0
```

Comment:
`absorbed_extracts_num` would be a bit more intuitive, or just `absorbed_extracts`